codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -334,18 +334,16 @@ public final int numChildren() {
         }
 
         @Override
-        public final Collection<? extends Http2Stream> children() {
-            return children.values();
-        }
-
-        @Override
-        public final boolean hasChild(int streamId) {
-            return child(streamId) != null;
-        }
-
-        @Override
-        public final Http2Stream child(int streamId) {
-            return children.get(streamId);
+        public Http2Stream forEachChild(Http2StreamVisitor visitor) throws Http2Exception {
+            Http2Stream resultStream = null;
+            for (IntObjectHashMap.Entry<DefaultStream> entry : children.entries()) {
+                Http2Stream stream = entry.value();
+                if (!visitor.visit(stream)) {
+                    resultStream = stream;
```

Comment:
You were probably just staying consistent with my code below...but I'm wondering can we just `return resultStream;` here and get ride of the temp variable.  Then also update the below method for consistency...my fault again :)